problem-2.py

```diff
@@ -0,0 +1,43 @@
+"""
+Problem: https://leetcode.com/problems/pascals-triangle-ii/
+
+Approach :
+
+1. we use 2 lists, cur and prev which are empty initialy
+2. we iterate till the rowIndex, if rowIndex is 0, we make prev = [1] and if rowIndex is 1, we make prev =[1,1] and continue
+3. one we are at index 2 we calcualte the cur list as follows:
+    a.append 1 to curr list 
+    b. while our prev list is empty, we append the sum of pairs of prev list to cur list
+    c. append 1 to cur list
+4. the we make prev=curr and cur = [] so that we can calcualte the next cur list based on the prev list
+
+"""
+
+# Time Complexity : O(n2)
+# Space Complexity : O(n)
```

Comment:
O(n^2)

Reply:
can you please help me with understanding how it is O(n^2)

Reply:
Do we use additional storage for returning the index?